File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/spot/SpotWalletApi.java
package com.bitget.openapi.api.spot;

import com.bitget.openapi.dto.request.spot.SpotSubTransferReq;
import com.bitget.openapi.dto.request.spot.SpotTransferReq;
import com.bitget.openapi.dto.request.spot.SpotWithdrawalInnerReq;
import com.bitget.openapi.dto.request.spot.SpotWithdrawalReq;
import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface SpotWalletApi {

    /**
     * transfer
     * @param body
     * @return ResponseResult
     */
    @POST("/api/spot/v1/wallet/transfer")
    Call<ResponseResult> transfer(@Body SpotTransferReq body);


    /**
     * subTransfer
     * @param body
     * @return ResponseResult
     */
    @POST("/api/spot/v1/wallet/subTransfer")
    Call<ResponseResult> subTransfer(@Body SpotSubTransferReq body);

    /**
     * get deposit address
     * @return ResponseResult
     */
    @GET("/api/spot/v1/wallet/deposit-address")
    Call<ResponseResult> depositAddress(@Query("coin")String coin,
                                        @Query("chain")String chain);

    /**
     * withdrawal
     * @return ResponseResult
     */
    @POST("/api/spot/v1/wallet/withdrawal")
    Call<ResponseResult> withdrawal(@Body SpotWithdrawalReq body);

    /**
     * withdrawal-inner
     * @return ResponseResult
     */
    @POST("/api/spot/v1/wallet/withdrawal-inner")
    Call<ResponseResult> withdrawalInner(@Body SpotWithdrawalInnerReq body);


    /**
     * get withdrawal record list
     * @return ResponseResult
     */
    @GET("/api/spot/v1/wallet/withdrawal-list")
    Call<ResponseResult> withdrawalList(@Query("coin")String coin,
                                        @Query("startTime") String startTime,
                                        @Query("endTime") String endTime,
                                        @Query("pageNo") Integer pageNo,
                                        @Query("pageSize") Integer pageSize);

    /**
     * get deposit record list
     * @return ResponseResult
     */
    @GET("/api/spot/v1/wallet/deposit-list")
    Call<ResponseResult> depositList(@Query("coin")String coin,
                                        @Query("startTime") String startTime,
                                        @Query("endTime") String endTime,
                                        @Query("pageNo") Integer pageNo,
                                        @Query("pageSize") Integer pageSize);


}

File: bitget-node-sdk-api/src/lib/v1/SpotWalletApi.ts
import {BaseApi} from '../BaseApi';

export class SpotWalletApi extends BaseApi {

    transfer(qsOrBody: object) {
        const url = '/api/spot/v1/wallet/transfer';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    depositAddress(qsOrBody: object) {
        const url = '/api/spot/v1/wallet/deposit-address';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    withdrawal(qsOrBody: object) {
        const url = '/api/spot/v1/wallet/withdrawal';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    withdrawalRecords(qsOrBody: object) {
        const url = '/api/spot/v1/wallet/withdrawal-list';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    depositRecords(qsOrBody: object) {
        const url = '/api/spot/v1/wallet/deposit-list';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }
}